### PR TITLE
Add postInstructions

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: ts/src/program/context.ts [hunk 1]
```diff
@@ -33,6 +33,12 @@ export type Context<A extends Accounts = Accounts> = {
    */
   instructions?: TransactionInstruction[];
 
+  /**
+   * Instructions to run *after* a given method. Often this is used, for
+   * example to close accounts after to executing a method.
+   */
+  postInstructions?: TransactionInstruction[];
```
Comment (armaniferrante) [inline ts/src/program/context.ts:40]: Want to add `preInstructions` for symmetry and a test?

Comment (SolanaMonkeyBusiness) [reply ts/src/program/context.ts:47]: Added preInstructions 
I only see 1 test file in the repo. Is there some idl mocking value to use for testing ?

File: ts/src/program/namespace/transaction.ts [hunk 2]
```diff
@@ -16,8 +16,13 @@ export default class TransactionFactory {
     const txFn: TransactionFn<IDL, I> = (...args): Transaction => {
       const [, ctx] = splitArgsAndCtx(idlIx, [...args]);
       const tx = new Transaction();
+      if(ctx.preInstructions && ctx.preInstructions.length && ctx.instructions && ctx.instructions.length){
```
Comment (armaniferrante) [inline ts/src/program/namespace/transaction.ts:19]:
```suggestion
      if (ctx.preInstructions && ctx.instructions) {
```

Is the `.length` needed here?

File: ts/src/program/context.ts (hunk 1 above)
Comment (armaniferrante) [reply ts/src/program/context.ts:47]: End to end tests are the easiest thing to do. Can we add a CHANGELOG entry as well.

Comment (SolanaMonkeyBusiness) [reply ts/src/program/context.ts:47]: Added 3 tests

**Comment by armaniferrante** [comment]
LGTM. Need to run `yarn lint:fix` for the build to pass.

**Comment by SolanaMonkeyBusiness** [comment]
> LGTM. Need to run `yarn lint:fix` for the build to pass.

Done ser

**Pull request merged**